[Bug Report] Unable to build iOS & package:flutter_ffmpeg_kit compatibility on Android

Command:
```
flutter build ios --release -t ./lib/main.dart --no-codesign
```

Logs for iOS:
```
ERROR: No pubspec.lock was found: ENV["PWD"] = "/"
[!] Failed to load 'media_kit_native_event_loop' podspec: 
[!] Invalid `media_kit_native_event_loop.podspec` file: ERROR: No pubspec.lock was found: ENV["PWD"] = "/".

 #  from /Users/runner/work/1/s/ios/.symlinks/plugins/media_kit_native_event_loop/ios/media_kit_native_event_loop.podspec:23
 #  -------------------------------------------
 #    if pubspec_lock_path == ''
 >      abort(
 #        sprintf('ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"', ENV["PWD"])
 #  -------------------------------------------

##[error]The process '/usr/local/lib/ruby/gems/3.0.0/bin/pod' failed with exit code 1
##[error]The 'pod' command failed with error: The process '/usr/local/lib/ruby/gems/3.0.0/bin/pod' failed with exit code 1
```

Command:
```
flutter build apk --release -t lib/main.dart 
```

Logs for Android:
```
* What went wrong:
Execution failed for task ':app:mergeReleaseNativeLibs'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.MergeJavaResWorkAction
   > 2 files found with path 'lib/x86/libavcodec.so' from inputs:
      - /Users/runner/.gradle/caches/transforms-3/179932a1e5526170ae637b254bf3c84e/transformed/jetified-video-x86.jar
      - /Users/runner/.gradle/caches/transforms-3/5f839fdc7e39313bb907a21151c1d4b2/transformed/jetified-ffmpeg-kit-min-gpl-5.1/jni
     If you are using jniLibs and CMake IMPORTED targets, see
     https://developer.android.com/r/tools/jniLibs-vs-imported-targets

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8m 1s
Running Gradle task 'assembleRelease'...                          481.5s
```

Flutter:
```
2023-04-25T05:30:26.0766540Z [!] Flutter (Channel unknown, 3.7.11, on macOS 12.6.5 21G531 darwin-x64, locale en)
2023-04-25T05:30:26.0785230Z     ! Flutter version 3.7.11 on channel unknown at /Users/runner/dev/flutter
2023-04-25T05:30:26.0792570Z       Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
2023-04-25T05:30:26.0793730Z       If that doesn't fix the issue, reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
2023-04-25T05:30:26.0797720Z     ! Unknown upstream repository.
2023-04-25T05:30:26.0798680Z       Reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
2023-04-25T05:30:33.7288570Z [!] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
2023-04-25T05:30:33.7290080Z     ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses
2023-04-25T05:30:33.7296680Z [✓] Xcode - develop for iOS and macOS (Xcode 14.2)
2023-04-25T05:30:33.7297590Z [✓] Chrome - develop for the web
2023-04-25T05:30:33.7298040Z [!] Android Studio (not installed)
2023-04-25T05:30:34.7372210Z [✓] Connected device (2 available)
2023-04-25T05:30:34.7374930Z [✓] HTTP Host Availability
```

Dependencies:
```
media_kit: ^0.0.6
media_kit_video: ^0.0.8
media_kit_native_event_loop: ^1.0.3
media_kit_libs_android_video: ^1.0.1
media_kit_libs_ios_video: ^1.0.4
```

@birros, i can confirm that my script successfully finds the `pubspec.lock`:
```
current_dir "/Users/runner/work/1/s/ios/.symlinks/plugins/media_kit_video/ios/media_kit_video.podspec"
Check path "/Users/runner/work/1/s/pubspec.lock"
File exist. Set pubspec_lock_path: "/Users/runner/work/1/s/pubspec.lock"
```

But then a command is launched from the MS App Center:

```
/usr/local/lib/ruby/gems/3.0.0/bin/pod install --repo-update
```

And the `pwd` path changes to: `"/Users/runner/work/1/s/ios/.symlinks/plugins/media_kit_video/ios/media_kit_video.podspec"`

```
2023-05-04T13:42:48.3140850Z ##[section]Starting: Pod install
2023-05-04T13:42:48.3158800Z ==============================================================================
2023-05-04T13:42:48.3158960Z Task         : CocoaPods
2023-05-04T13:42:48.3159050Z Description  : Install CocoaPods dependencies for Swift and Objective-C Cocoa projects
2023-05-04T13:42:48.3159230Z Version      : 0.220.0
2023-05-04T13:42:48.3159310Z Author       : Microsoft Corporation
2023-05-04T13:42:48.3159440Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/package/cocoapods
2023-05-04T13:42:48.3159570Z ==============================================================================
2023-05-04T13:42:48.6403340Z [command]/usr/local/lib/ruby/gems/3.0.0/bin/pod --version
2023-05-04T13:42:50.7883030Z 1.12.1
2023-05-04T13:42:50.7999290Z [command]/usr/local/lib/ruby/gems/3.0.0/bin/pod install --repo-update
2023-05-04T13:42:52.8624370Z Updating local specs repositories
2023-05-04T13:42:53.0222980Z Analyzing dependencies
2023-05-04T13:42:53.1283550Z make: Nothing to be done for `all'.
2023-05-04T13:42:53.1546170Z make: Nothing to be done for `all'.
2023-05-04T13:42:53.1562700Z current_dir "/Users/runner/work/1/s/ios/.symlinks/plugins/media_kit_video/ios/media_kit_video.podspec"
2023-05-04T13:42:53.1564190Z ERROR: No pubspec.lock was found: ENV["PWD"] = "/"
2023-05-04T13:42:53.1573980Z [!] Failed to load 'media_kit_video' podspec: 
2023-05-04T13:42:53.1574770Z [!] Invalid `media_kit_video.podspec` file: ERROR: No pubspec.lock was found: ENV["PWD"] = "/".
2023-05-04T13:42:53.1574950Z 
2023-05-04T13:42:53.1575230Z  #  from /Users/runner/work/1/s/ios/.symlinks/plugins/media_kit_video/ios/media_kit_video.podspec:27
2023-05-04T13:42:53.1575660Z  #  -------------------------------------------
2023-05-04T13:42:53.1575920Z  #    if pubspec_lock_path == ''
2023-05-04T13:42:53.1576120Z  >      abort(
2023-05-04T13:42:53.1576480Z  #        sprintf('ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"', ENV['PWD'])
2023-05-04T13:42:53.1576870Z  #  -------------------------------------------
2023-05-04T13:42:53.1799300Z ##[error]The process '/usr/local/lib/ruby/gems/3.0.0/bin/pod' failed with exit code 1
2023-05-04T13:42:53.1811520Z ##[error]The 'pod' command failed with error: The process '/usr/local/lib/ruby/gems/3.0.0/bin/pod' failed with exit code 1
2023-05-04T13:42:53.1815310Z ##[section]Finishing: Pod install
```

--- a/media_kit_native_event_loop/common/darwin/Podspec/media_kit_utils.rb
+++ b/media_kit_native_event_loop/common/darwin/Podspec/media_kit_utils.rb
@@ -1,0 +1,98 @@
+# `MediaKitUtils` is duplicated across:
+# - `media_kit_video`
+# - `media_kit_native_event_loop`
+class MediaKitUtils
+  attr_accessor :libs_found, :libs_package
+
+  module Platform
+    IOS   ||= 'ios'
+    MACOS ||= 'macos'
+  end
+
+  module Type
+    AUDIO ||= 'audio'
+    VIDEO ||= 'video'
+  end
+
+  def initialize(platform)
+    raise "Expecting '#{Platform::IOS}' or '#{Platform::MACOS}': platform = #{platform}" if
+      !(platform == Platform::IOS || platform == Platform::MACOS)
+
+    # Find the nearest path to `pubspec.lock` using `PWD` env var set by
+    # `cocoapods`
+    current_dir       = ENV['PWD'] || '/'
+    pubspec_lock_path = MediaKitUtils::find_nearest_pubspec_lock(current_dir)
+
+    # If not found try again using `PWD_FALLBACK` env var manually set,
+    # `PWD_FALLBACK` must point to app folder or any child folder
+    if pubspec_lock_path == ''
+      current_dir       = ENV['PWD_FALLBACK'] || '/'
+      pubspec_lock_path = MediaKitUtils::find_nearest_pubspec_lock(current_dir)
+    end
+
+    # Abort if no `pubspec.lock` was found
+    if pubspec_lock_path == ''
+      abort(
+        sprintf(
+          'media_kit: ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"',
+          ENV["PWD"]
+        )
+      )
+    end
+
+    # Load packages from `pubspec.lock`
+    pubspec_lock          = YAML.load_file(pubspec_lock_path)
+    packages              = pubspec_lock['packages']
+    
+    # Checks for `media_kit_libs_***` in `pubspec.lock`
+    libs_count    = 0
+    @libs_package = ''
+    Type.constants.each do |constant|
+      type    = Type.const_get(constant)
+      package = sprintf('media_kit_libs_%s_%s', platform, type)
+      if packages.keys.include?(package)
+        libs_count    += 1
+        @libs_package = package
+
+        puts sprintf('media_kit: INFO: package:%s found', package)
+      end
+    end
+
+    # Abort if multiple `media_kit_libs_***` was found
+    if libs_count > 1
+      abort(
+        sprintf(
+          'media_kit: ERROR: package:media_kit_libs_%s_*** must be uniq',
+          platform,
+        )
+      )
+    end
+
+    @libs_found = libs_count > 0
+
+    # Warn if no `media_kit_libs_*` was found
+    if !@libs_found
+      warn(
+        sprintf(
+          'media_kit: WARNING: package:media_kit_libs_%s_*** not found',
+          platform
+        )
+      )
+    end
+  end
+
+  # Looks for the nearest `pubspec.lock` by recursively ascending through the
+  # parent folders
+  def self.find_nearest_pubspec_lock(current_dir)
+    while current_dir != '/'
+      path = File.join(current_dir, 'pubspec.lock')
+      if File.exist?(path)
+        return path
+      else
+        current_dir = File.expand_path('..', current_dir)
+      end
+    end
+
+    return ''
+  end
+end

--- a/media_kit_native_event_loop/ios/media_kit_native_event_loop.podspec
+++ b/media_kit_native_event_loop/ios/media_kit_native_event_loop.podspec
@@ -2,51 +2,15 @@
 # To learn more about a Podspec see http://guides.cocoapods.org/syntax/podspec.html.
 # Run `pod lib lint media_kit_native_event_loop.podspec` to validate before publishing.
 #
+
+require_relative '../common/darwin/Podspec/media_kit_utils.rb'
+
 Pod::Spec.new do |s|
   # Setup required files
   system("make -C ../common/darwin")
 
-  # Find the nearest path to `pubspec.lock`
-  current_dir       = ENV['PWD']
-  pubspec_lock_path = ''
-  while pubspec_lock_path == '' && current_dir != '/'
-    path = File.join(current_dir, 'pubspec.lock')
-    if File.exist?(path)
-      pubspec_lock_path = path
-    else
-      current_dir = File.expand_path('..', current_dir)
-    end
-  end
-
-  # Fail if no `pubspec.lock` was found
-  if pubspec_lock_path == ''
-    abort(
-      sprintf('ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"', ENV["PWD"])
-    )
-  end
-
-  # Checks for `media_kit_libs_*` in `pubspec.lock`
-  pubspec_lock           = YAML.load_file(pubspec_lock_path)
-  packages               = pubspec_lock['packages']
-  libs_audio_dep_found   = packages.keys.include?('media_kit_libs_ios_audio')
-  libs_video_dep_found   = packages.keys.include?('media_kit_libs_ios_video')
-  libs_dep_found         = libs_audio_dep_found || libs_video_dep_found
-
-  # Warn if no `media_kit_libs_*` is found
-  if !libs_dep_found
-    warn("media_kit: WARNING: package:media_kit_libs_*** not found")
-  end
-
-  # Define paths to frameworks dir
-  framework_search_paths_iphoneos        = ''
-  framework_search_paths_iphonesimulator = ''
-  if libs_audio_dep_found
-    framework_search_paths_iphoneos        = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_audio/ios/Frameworks/MPV.xcframework/ios-arm64'
-    framework_search_paths_iphonesimulator = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_audio/ios/Frameworks/MPV.xcframework/ios-arm64_x86_64-simulator'
-  elsif libs_video_dep_found
-    framework_search_paths_iphoneos        = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_video/ios/Frameworks/MPV.xcframework/ios-arm64'
-    framework_search_paths_iphonesimulator = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_video/ios/Frameworks/MPV.xcframework/ios-arm64_x86_64-simulator'
-  end
+  # Initialize `MediaKitUtils`
+  mku = MediaKitUtils.new(MediaKitUtils::Platform::IOS)
 
   s.name             = 'media_kit_native_event_loop'
   s.version          = '1.0.0'
@@ -65,7 +29,11 @@ Pod::Spec.new do |s|
   s.source           = { :path => '.' }
   s.dependency         'Flutter'
 
-  if libs_dep_found
+  if mku.libs_found
+    # Define paths to frameworks dir
+    framework_search_paths_iphoneos        = sprintf('$(PROJECT_DIR)/../.symlinks/plugins/%s/ios/Frameworks/MPV.xcframework/ios-arm64', mku.libs_package)
+    framework_search_paths_iphonesimulator = sprintf('$(PROJECT_DIR)/../.symlinks/plugins/%s/ios/Frameworks/MPV.xcframework/ios-arm64_x86_64-simulator', mku.libs_package)
+
     s.source_files        = 'Classes/**/*'
     s.platform            = :ios, '13.0'
     s.swift_version       = '5.0'

--- a/media_kit_native_event_loop/macos/media_kit_native_event_loop.podspec
+++ b/media_kit_native_event_loop/macos/media_kit_native_event_loop.podspec
@@ -2,48 +2,15 @@
 # To learn more about a Podspec see http://guides.cocoapods.org/syntax/podspec.html.
 # Run `pod lib lint media_kit_native_event_loop.podspec` to validate before publishing.
 #
+
+require_relative '../common/darwin/Podspec/media_kit_utils.rb'
+
 Pod::Spec.new do |s|
   # Setup required files
   system("make -C ../common/darwin")
 
-  # Find the nearest path to `pubspec.lock`
-  current_dir       = ENV['PWD']
-  pubspec_lock_path = ''
-  while pubspec_lock_path == '' && current_dir != '/'
-    path = File.join(current_dir, 'pubspec.lock')
-    if File.exist?(path)
-      pubspec_lock_path = path
-    else
-      current_dir = File.expand_path('..', current_dir)
-    end
-  end
-
-  # Fail if no `pubspec.lock` was found
-  if pubspec_lock_path == ''
-    abort(
-      sprintf('ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"', ENV["PWD"])
-    )
-  end
-
-  # Checks for `media_kit_libs_*` in `pubspec.lock`
-  pubspec_lock           = YAML.load_file(pubspec_lock_path)
-  packages               = pubspec_lock['packages']
-  libs_audio_dep_found   = packages.keys.include?('media_kit_libs_macos_audio')
-  libs_video_dep_found   = packages.keys.include?('media_kit_libs_macos_video')
-  libs_dep_found         = libs_audio_dep_found || libs_video_dep_found
-
-  # Warn if no `media_kit_libs_*` is found
-  if !libs_dep_found
-    warn("media_kit: WARNING: package:media_kit_libs_*** not found")
-  end
-
-  # Define paths to frameworks dir
-  framework_search_paths_macosx = ''
-  if libs_audio_dep_found
-    framework_search_paths_macosx = '$(PROJECT_DIR)/../Flutter/ephemeral/.symlinks/plugins/media_kit_libs_macos_audio/macos/Frameworks/MPV.xcframework/macos-arm64_x86_64'
-  elsif libs_video_dep_found
-    framework_search_paths_macosx = '$(PROJECT_DIR)/../Flutter/ephemeral/.symlinks/plugins/media_kit_libs_macos_video/macos/Frameworks/MPV.xcframework/macos-arm64_x86_64'
-  end
+  # Initialize `MediaKitUtils`
+  mku = MediaKitUtils.new(MediaKitUtils::Platform::MACOS)
 
   s.name             = 'media_kit_native_event_loop'
   s.version          = '1.0.0'
@@ -62,7 +29,10 @@ Pod::Spec.new do |s|
   s.source           = { :path => '.' }
   s.dependency         'FlutterMacOS'
 
-  if libs_dep_found
+  if mku.libs_found
+    # Define paths to frameworks dir
+    framework_search_paths_macosx = sprintf('$(PROJECT_DIR)/../Flutter/ephemeral/.symlinks/plugins/%s/macos/Frameworks/MPV.xcframework/macos-arm64_x86_64', mku.libs_package)
+
     s.source_files        = 'Classes/**/*'
     s.platform            = :osx, '10.9'
     s.swift_version       = '5.0'

--- a/media_kit_video/common/darwin/Podspec/media_kit_utils.rb
+++ b/media_kit_video/common/darwin/Podspec/media_kit_utils.rb
@@ -1,0 +1,98 @@
+# `MediaKitUtils` is duplicated across:
+# - `media_kit_video`
+# - `media_kit_native_event_loop`
+class MediaKitUtils
+  attr_accessor :libs_found, :libs_package
+
+  module Platform
+    IOS   ||= 'ios'
+    MACOS ||= 'macos'
+  end
+
+  module Type
+    AUDIO ||= 'audio'
+    VIDEO ||= 'video'
+  end
+
+  def initialize(platform)
+    raise "Expecting '#{Platform::IOS}' or '#{Platform::MACOS}': platform = #{platform}" if
+      !(platform == Platform::IOS || platform == Platform::MACOS)
+
+    # Find the nearest path to `pubspec.lock` using `PWD` env var set by
+    # `cocoapods`
+    current_dir       = ENV['PWD'] || '/'
+    pubspec_lock_path = MediaKitUtils::find_nearest_pubspec_lock(current_dir)
+
+    # If not found try again using `PWD_FALLBACK` env var manually set,
+    # `PWD_FALLBACK` must point to app folder or any child folder
+    if pubspec_lock_path == ''
+      current_dir       = ENV['PWD_FALLBACK'] || '/'
+      pubspec_lock_path = MediaKitUtils::find_nearest_pubspec_lock(current_dir)
+    end
+
+    # Abort if no `pubspec.lock` was found
+    if pubspec_lock_path == ''
+      abort(
+        sprintf(
+          'media_kit: ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"',
+          ENV["PWD"]
+        )
+      )
+    end
+
+    # Load packages from `pubspec.lock`
+    pubspec_lock          = YAML.load_file(pubspec_lock_path)
+    packages              = pubspec_lock['packages']
+    
+    # Checks for `media_kit_libs_***` in `pubspec.lock`
+    libs_count    = 0
+    @libs_package = ''
+    Type.constants.each do |constant|
+      type    = Type.const_get(constant)
+      package = sprintf('media_kit_libs_%s_%s', platform, type)
+      if packages.keys.include?(package)
+        libs_count    += 1
+        @libs_package = package
+
+        puts sprintf('media_kit: INFO: package:%s found', package)
+      end
+    end
+
+    # Abort if multiple `media_kit_libs_***` was found
+    if libs_count > 1
+      abort(
+        sprintf(
+          'media_kit: ERROR: package:media_kit_libs_%s_*** must be uniq',
+          platform,
+        )
+      )
+    end
+
+    @libs_found = libs_count > 0
+
+    # Warn if no `media_kit_libs_*` was found
+    if !@libs_found
+      warn(
+        sprintf(
+          'media_kit: WARNING: package:media_kit_libs_%s_*** not found',
+          platform
+        )
+      )
+    end
+  end
+
+  # Looks for the nearest `pubspec.lock` by recursively ascending through the
+  # parent folders
+  def self.find_nearest_pubspec_lock(current_dir)
+    while current_dir != '/'
+      path = File.join(current_dir, 'pubspec.lock')
+      if File.exist?(path)
+        return path
+      else
+        current_dir = File.expand_path('..', current_dir)
+      end
+    end
+
+    return ''
+  end
+end

--- a/media_kit_video/ios/media_kit_video.podspec
+++ b/media_kit_video/ios/media_kit_video.podspec
@@ -2,51 +2,15 @@
 # To learn more about a Podspec see http://guides.cocoapods.org/syntax/podspec.html.
 # Run `pod lib lint media_kit_video.podspec` to validate before publishing.
 #
+
+require_relative '../common/darwin/Podspec/media_kit_utils.rb'
+
 Pod::Spec.new do |s|
   # Setup required files
   system("make -C ../common/darwin HEADERS_DESTDIR=\"$(pwd)/Headers\"")
 
-  # Find the nearest path to `pubspec.lock`
-  current_dir       = ENV['PWD']
-  pubspec_lock_path = ''
-  while pubspec_lock_path == '' && current_dir != '/'
-    path = File.join(current_dir, 'pubspec.lock')
-    if File.exist?(path)
-      pubspec_lock_path = path
-    else
-      current_dir = File.expand_path('..', current_dir)
-    end
-  end
-
-  # Fail if no `pubspec.lock` was found
-  if pubspec_lock_path == ''
-    abort(
-      sprintf('ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"', ENV["PWD"])
-    )
-  end
-
-  # Checks for `media_kit_libs_*` in `pubspec.lock`
-  pubspec_lock           = YAML.load_file(pubspec_lock_path)
-  packages               = pubspec_lock['packages']
-  libs_audio_dep_found   = packages.keys.include?('media_kit_libs_ios_audio')
-  libs_video_dep_found   = packages.keys.include?('media_kit_libs_ios_video')
-  libs_dep_found         = libs_audio_dep_found || libs_video_dep_found
-
-  # Warn if no `media_kit_libs_*` is found
-  if !libs_dep_found
-    warn("media_kit: WARNING: package:media_kit_libs_*** not found")
-  end
-
-  # Define paths to frameworks dir
-  framework_search_paths_iphoneos        = ''
-  framework_search_paths_iphonesimulator = ''
-  if libs_audio_dep_found
-    framework_search_paths_iphoneos        = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_audio/ios/Frameworks/MPV.xcframework/ios-arm64'
-    framework_search_paths_iphonesimulator = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_audio/ios/Frameworks/MPV.xcframework/ios-arm64_x86_64-simulator'
-  elsif libs_video_dep_found
-    framework_search_paths_iphoneos        = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_video/ios/Frameworks/MPV.xcframework/ios-arm64'
-    framework_search_paths_iphonesimulator = '$(PROJECT_DIR)/../.symlinks/plugins/media_kit_libs_ios_video/ios/Frameworks/MPV.xcframework/ios-arm64_x86_64-simulator'
-  end
+  # Initialize `MediaKitUtils`
+  mku = MediaKitUtils.new(MediaKitUtils::Platform::IOS)
 
   s.name             = 'media_kit_video'
   s.version          = '0.0.1'
@@ -63,7 +27,11 @@ Pod::Spec.new do |s|
   s.swift_version    = '5.0'
   s.dependency         'Flutter'
   
-  if libs_dep_found
+  if mku.libs_found
+    # Define paths to frameworks dir
+    framework_search_paths_iphoneos        = sprintf('$(PROJECT_DIR)/../.symlinks/plugins/%s/ios/Frameworks/MPV.xcframework/ios-arm64', mku.libs_package)
+    framework_search_paths_iphonesimulator = sprintf('$(PROJECT_DIR)/../.symlinks/plugins/%s/ios/Frameworks/MPV.xcframework/ios-arm64_x86_64-simulator', mku.libs_package)
+
     s.source_files        = 'Classes/plugin/**/*.swift', 'Headers/**/*.h'
     s.pod_target_xcconfig = {
       'DEFINES_MODULE'                               => 'YES',

--- a/media_kit_video/macos/media_kit_video.podspec
+++ b/media_kit_video/macos/media_kit_video.podspec
@@ -2,48 +2,15 @@
 # To learn more about a Podspec see http://guides.cocoapods.org/syntax/podspec.html.
 # Run `pod lib lint media_kit_video.podspec` to validate before publishing.
 #
+
+require_relative '../common/darwin/Podspec/media_kit_utils.rb'
+
 Pod::Spec.new do |s|
   # Setup required files
   system("make -C ../common/darwin HEADERS_DESTDIR=\"$(pwd)/Headers\"")
 
-  # Find the nearest path to `pubspec.lock`
-  current_dir       = ENV['PWD']
-  pubspec_lock_path = ''
-  while pubspec_lock_path == '' && current_dir != '/'
-    path = File.join(current_dir, 'pubspec.lock')
-    if File.exist?(path)
-      pubspec_lock_path = path
-    else
-      current_dir = File.expand_path('..', current_dir)
-    end
-  end
-
-  # Fail if no `pubspec.lock` was found
-  if pubspec_lock_path == ''
-    abort(
-      sprintf('ERROR: No pubspec.lock was found: ENV["PWD"] = "%s"', ENV["PWD"])
-    )
-  end
-
-  # Checks for `media_kit_libs_*` in `pubspec.lock`
-  pubspec_lock           = YAML.load_file(pubspec_lock_path)
-  packages               = pubspec_lock['packages']
-  libs_audio_dep_found   = packages.keys.include?('media_kit_libs_macos_audio')
-  libs_video_dep_found   = packages.keys.include?('media_kit_libs_macos_video')
-  libs_dep_found         = libs_audio_dep_found || libs_video_dep_found
-
-  # Warn if no `media_kit_libs_*` is found
-  if !libs_dep_found
-    warn("media_kit: WARNING: package:media_kit_libs_*** not found")
-  end
-
-  # Define paths to frameworks dir
-  framework_search_paths_macosx = ''
-  if libs_audio_dep_found
-    framework_search_paths_macosx = '$(PROJECT_DIR)/../Flutter/ephemeral/.symlinks/plugins/media_kit_libs_macos_audio/macos/Frameworks/MPV.xcframework/macos-arm64_x86_64'
-  elsif libs_video_dep_found
-    framework_search_paths_macosx = '$(PROJECT_DIR)/../Flutter/ephemeral/.symlinks/plugins/media_kit_libs_macos_video/macos/Frameworks/MPV.xcframework/macos-arm64_x86_64'
-  end
+  # Initialize `MediaKitUtils`
+  mku = MediaKitUtils.new(MediaKitUtils::Platform::MACOS)
 
   s.name             = 'media_kit_video'
   s.version          = '0.0.1'
@@ -60,7 +27,10 @@ Pod::Spec.new do |s|
   s.swift_version    = '5.0'
   s.dependency         'FlutterMacOS'
 
-  if libs_dep_found
+  if mku.libs_found
+    # Define paths to frameworks dir
+    framework_search_paths_macosx = sprintf('$(PROJECT_DIR)/../Flutter/ephemeral/.symlinks/plugins/%s/macos/Frameworks/MPV.xcframework/macos-arm64_x86_64', mku.libs_package)
+
     s.source_files        = 'Classes/plugin/**/*.swift', 'Headers/**/*.h'
     s.pod_target_xcconfig = {
       'DEFINES_MODULE'                      => 'YES',

